pysmt/walkers/generic.py

```diff
@@ -30,80 +74,44 @@ def __init__(self, env=None):
-        self.functions[op.REAL_CONSTANT] = self.walk_real_constant
-        self.functions[op.BOOL_CONSTANT] = self.walk_bool_constant
-        self.functions[op.INT_CONSTANT] = self.walk_int_constant
-        self.functions[op.PLUS] = self.walk_plus
-        self.functions[op.MINUS] = self.walk_minus
-        self.functions[op.TIMES] = self.walk_times
-        self.functions[op.POW] = self.walk_pow
-        self.functions[op.EQUALS] = self.walk_equals
-        self.functions[op.LE] = self.walk_le
-        self.functions[op.LT] = self.walk_lt
-        self.functions[op.ITE] = self.walk_ite
-        self.functions[op.TOREAL] = self.walk_toreal
-
-        self.functions[op.BV_CONSTANT] = self.walk_bv_constant
-        self.functions[op.BV_CONCAT] = self.walk_bv_concat
-        self.functions[op.BV_EXTRACT] = self.walk_bv_extract
-        self.functions[op.BV_NOT] = self.walk_bv_not
-        self.functions[op.BV_AND] = self.walk_bv_and
-        self.functions[op.BV_OR] = self.walk_bv_or
-        self.functions[op.BV_XOR] = self.walk_bv_xor
-        self.functions[op.BV_ULT] = self.walk_bv_ult
-        self.functions[op.BV_ULE] = self.walk_bv_ule
-        self.functions[op.BV_NEG] = self.walk_bv_neg
-        self.functions[op.BV_ADD] = self.walk_bv_add
-        self.functions[op.BV_SUB] = self.walk_bv_sub
-        self.functions[op.BV_MUL] = self.walk_bv_mul
-        self.functions[op.BV_UDIV] = self.walk_bv_udiv
-        self.functions[op.BV_UREM] = self.walk_bv_urem
-        self.functions[op.BV_LSHL] = self.walk_bv_lshl
-        self.functions[op.BV_LSHR] = self.walk_bv_lshr
-        self.functions[op.BV_ROL] = self.walk_bv_rol
-        self.functions[op.BV_ROR] = self.walk_bv_ror
-        self.functions[op.BV_ZEXT] = self.walk_bv_zext
-        self.functions[op.BV_SEXT] = self.walk_bv_sext
-        self.functions[op.BV_SLT] = self.walk_bv_slt
-        self.functions[op.BV_SLE] = self.walk_bv_sle
-        self.functions[op.BV_COMP] = self.walk_bv_comp
-        self.functions[op.BV_SDIV] = self.walk_bv_sdiv
-        self.functions[op.BV_SREM] = self.walk_bv_srem
-        self.functions[op.BV_ASHR] = self.walk_bv_ashr
-
-        self.functions[op.ARRAY_SELECT] = self.walk_array_select
-        self.functions[op.ARRAY_STORE] = self.walk_array_store
-        self.functions[op.ARRAY_VALUE] = self.walk_array_value
-
-        self.functions[op.DIV] = self.walk_div
-        self.functions[op.ALGEBRAIC_CONSTANT] = self.walk_algebraic_constant
-
-        undefined_types = set(op.ALL_TYPES) - set(self.functions.keys())
-        assert len(undefined_types) == 0, \
-            "The following types are not defined in the generic walker: {%s}" % \
-            (", ".join(op.op_to_str(u) for u in undefined_types))
-
+        for o in op.all_types():
+            try:
+                # getattr will raise an AttributeError exception if a
+                # method does not exist
+                self.functions[o] = getattr(self, nt_to_fun(o))
```

Comment:
The ``functions`` dictionary is there **only** for backwards compatibility, for this reason it might not be identical (due to the nature of the instance-based walkers).
 
If you use it (via ``set_function``), you will get a deprecation warning. In a future version (0.8.0?) we will remove it, and the class-based walker will be the only way to define a walker.